src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,34 @@
+import { splitProps } from '~/utils/splitProps';
+import {Box, type BoxProps} from '../Box/Box';
+import {textinput, type TextinputVariantProps } from '@styled-system/recipes';
+import { cx } from '@styled-system/css';
+
+
+export type TextInputProps = Omit<BoxProps, keyof TextinputVariantProps> & TextinputVariantProps &{
+    disabled?: boolean,
+    error?: boolean,
+}
+
+export const TextInput: React.FC<TextInputProps> = (
+    {
+        size,
+        error,
+        autoSize = false,
+        ...props
+    }: TextInputProps
+) => {
+
+    const [ className, otherProps] = splitProps(props);
+    return(
+        <Box
+        as="input"
+        data-error={error ? '' : undefined} 
```

Comment:
Change this to check for error and if it exists, add data-error attribute
```suggestion
        error ? data-error : undefined
```